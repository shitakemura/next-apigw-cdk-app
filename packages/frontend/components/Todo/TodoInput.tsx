import { Button, HStack, Input } from "@chakra-ui/react";
import { useState } from "react";
import { useTodos } from "../../hooks/useTodosContext";

export const TodoInput = () => {
  const [titleInput, setTitleInput] = useState("");
  const { isLoading, createTodo } = useTodos();

  return (
    <HStack spacing={6}>
      <Input
        borderColor='green.500'
        borderWidth={2}
        height={12}
        width={400}
        value={titleInput}
        onChange={(e) => setTitleInput(e.target.value)}
      />
      <Button
        paddingX={8}
        colorScheme='green'
        disabled={!titleInput}
        isLoading={isLoading}
        onClick={() => createTodo({ title: titleInput })}>
        Add Todo
      </Button>
    </HStack>
  );
};
