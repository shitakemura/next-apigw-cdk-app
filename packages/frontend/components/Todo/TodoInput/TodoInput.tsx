import { Button, HStack, Input } from "@chakra-ui/react";
import { useCallback, useMemo, useState } from "react";
import { useTodos } from "../../../hooks/useTodosContext";

export const TodoInput = () => {
  const [titleInput, setTitleInput] = useState("");
  const { isLoading, createTodo } = useTodos();
  const clearTitleInput = useCallback(() => setTitleInput(""), []);

  const isPostLoading = useMemo(() => {
    return (
      (isLoading && isLoading.type === "POST" && isLoading.value === true) ??
      false
    );
  }, [isLoading]);

  return (
    <HStack spacing={6}>
      <Input
        borderColor='green.500'
        borderWidth={2}
        height={12}
        width={400}
        value={titleInput}
        onChange={(e) => setTitleInput(e.target.value)}
      />
      <Button
        paddingX={8}
        colorScheme='green'
        disabled={!titleInput}
        isLoading={isPostLoading}
        onClick={() => {
          createTodo({ title: titleInput });
          clearTitleInput();
        }}>
        Add Todo
      </Button>
    </HStack>
  );
};
