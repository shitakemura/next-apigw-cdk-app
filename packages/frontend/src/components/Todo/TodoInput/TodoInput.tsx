import { Button, HStack, Input, useToast } from "@chakra-ui/react";
import { useCallback, useEffect, useState } from "react";
import { useCreateTodo } from "../../../hooks/todos/useCreateTodo";

export const TodoInput = () => {
  const [titleInput, setTitleInput] = useState("");
  const { createStatus, error, createTodo } = useCreateTodo();
  const clearTitleInput = useCallback(() => setTitleInput(""), []);
  const toast = useToast();

  useEffect(() => {
    if (error) {
      toast({ title: error.message, status: "error", isClosable: true });
    }
  }, [error, toast]);

  return (
    <HStack spacing={6}>
      <Input
        borderColor='green.500'
        borderWidth={2}
        height={12}
        width={400}
        value={titleInput}
        onChange={(e) => setTitleInput(e.target.value)}
      />
      <Button
        paddingX={8}
        colorScheme='green'
        disabled={!titleInput}
        isLoading={createStatus.isLoading}
        onClick={() => {
          createTodo({ title: titleInput });
          clearTitleInput();
        }}>
        Add Todo
      </Button>
    </HStack>
  );
};
