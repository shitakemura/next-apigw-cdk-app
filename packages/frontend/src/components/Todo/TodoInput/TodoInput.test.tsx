import { cleanup, render, screen } from "@testing-library/react";
import { TodoInput } from "./TodoInput";

describe("TodoInput", () => {
  test("初期表示", () => {
    const { asFragment } = render(<TodoInput />);
    expect(asFragment()).toMatchSnapshot();

    const input = screen.getByRole("textbox");
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue("");

    const button = screen.getByRole("button", { name: "Add Todo" });
    expect(button).toBeInTheDocument();
    expect(button).toBeDisabled();
  });
});
