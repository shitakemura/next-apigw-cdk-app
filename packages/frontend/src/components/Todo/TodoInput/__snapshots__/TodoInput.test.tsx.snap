// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput 初期表示 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-cxk74f"
  >
    <input
      class="chakra-input css-kggzbs"
      value=""
    />
    <button
      class="chakra-button css-1bu2fo6"
      disabled=""
      type="button"
    >
      Add Todo
    </button>
  </div>
</DocumentFragment>
`;
